pkg/controllers/networking/pod_controller.go

```diff
@@ -183,55 +184,68 @@ func (r *PodReconciler) selectNetwork(pod *corev1.Pod) (string, error) {
 	switch networkType {
 	case types.Underlay:
 		// try to get underlay network by node indexer
-		var networkList *networkingv1.NetworkList
-		var err error
-		if networkList, err = utils.ListNetworks(r, client.MatchingFields{IndexerFieldNode: pod.Spec.NodeName}); err != nil {
-			return "", fmt.Errorf("unable to list underlay network by indexer node: %v", err)
-		}
-		if len(networkList.Items) >= 1 {
-			return networkList.Items[0].GetName(), nil
+		underlayNetworkName, err := r.getNetworkByNodeNameIndexer(pod.Spec.NodeName)
+		if err != nil {
+			return "", fmt.Errorf("unable to get underlay network by node name indexer: %v", err)
 		}
 
-		// fall back to find underlay network by label selector
-		var underlayNetworkName string
-		if underlayNetworkName, err = utils.FindUnderlayNetworkForNodeName(r, pod.Spec.NodeName); err != nil {
-			return "", fmt.Errorf("unable to find underlay network for node %s", pod.Spec.NodeName)
-		}
-		if len(underlayNetworkName) == 0 {
-			return "", fmt.Errorf("no underlay network match node %s", pod.Spec.NodeName)
-		}
-		if !r.matchNetworkTypeInManager(underlayNetworkName, types.Underlay) {
-			return "", fmt.Errorf("network %s does not match type %q in manager", underlayNetworkName, types.Underlay)
+		if len(underlayNetworkName) != 0 {
+			if !r.matchNetworkTypeInManager(underlayNetworkName, types.Underlay) {
```

Comment:
can you restore these lines to 
```
if len(underlayNetworkName) == 0 {
    ...
}
if !r.matchNetworkTypeInManager(underlayNetworkName, types.Underlay) {
   ...
}
```

IMO, the nesting of several `if` is not very clear.

Reply:
Good idea. It's done.